update-generated.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+
+set -ex
+
+SOURCE_BRANCH=${1:-master}
+GENERATED_BRANCH=${2:-generated}
+
+git clean -f -d
+
+# checkout source branch
+git checkout "$SOURCE_BRANCH" && git submodule update
+tag=$(git rev-parse HEAD)
+
+# copy the branch content into source/
+rsync -a ./* source
+VERSIONS=$(sed -n 's/^VERSIONS[[:space:]]*=//p' source/Makefile)
+
+git checkout "$GENERATED_BRANCH" && git submodule update
+for i in *; do
```

Comment:
What about

```bash
shopt -s extglob
rm -rf !(common|sources)
```
?

Reply:
There is still and update symlink in root that points to common scripts for sake of convenience.

Using extended globbing for this is a good idea though, thanks!

Reply:
>There is still an update symlink in root that points to common scripts for sake of convenience.

That is for the postgresql POC I will base from this PR at least.